### PR TITLE
Dump pokemon stats to file

### PR DESCRIPTION
**Pull request by JoshYuen**
Dumps all your pokemon stats to a file in case you wish to look at them. Since the logger currently only shows top 20 of each category. 

Will add stats in a future PR.
image

### CODE REVIEW

**Comment by JoshYuen** [comment]
Wanted to make a few changes removing PR for now.

File: PoGo.NecroBot.Logic/Logging/Logger.cs [hunk 1]
```diff
@@ -30,6 +29,41 @@ private static void Log(string message)
         }
 
         /// <summary>
+        ///     This is used for dumping contents to a file stored in the Logs folder.
+        /// </summary>
+        /// <param name="data">Dumps the string data to the file</param>
+        /// <param name="filename">Filename to be used for naming the file.</param>
+        private static void DumpToFile(string data, string filename)
```
Comment (yamashi) [inline PoGo.NecroBot.Logic/Logging/Logger.cs]: This is actually unrelated to Logger

**Comment by JoshYuen** [comment]
I will move it out then to its own cs

**Comment by JoshYuen** [comment]
Good to go, moved it to its own namespace.

**Comment by GTXMike** [comment]
hope this soon goes to the releases

**Comment by 5andr0** [comment]
Please use GlobalSettings.ProfilePath instead of current dir to support multi user profiles

**Comment by JoshYuen** [comment]
Fixed to use the ProfilePath. Do note someone needs to update all the references to using the GetDirectory to use ProfilePath as well.

**Comment by qaqsd123** [comment]
This one did it really well. Even had hyper links to pokerater for all pokemon.

https://github.com/NecronomiconCoding/NecroBot/pull/462

**Comment by JoshYuen** [comment]
We can improve it to do those in the future, however I wouldn't depend on third party applications or links. Incase they ever get discontinued or deleted we rather not just give users a 404. The listing of all the information isn't a bad thing I just wanted the ability to dump all my pokemon so when I login I know which to evolve and level up myself.

**Comment by JoshYuen** [comment]
There is an issue with this PR actually you need to add a getter function for ProfilePath in FeroxRev. Maybe I should revert to older implementation until that gets updated?

**Comment by brandy** [comment]
:+1:

**Comment by 5andr0** [comment]
Why did you revert it? GlobalSettings.ConfigPath is public so you won't need a getter?

**Comment by NecronomiconCoding** [comment]
please update your changes with the current merge and i will be accepting that

**Comment by JoshYuen** [comment]
Updated, sorry was away from my computer.

**Comment by 5andr0** [comment]
Looks good, thanks for the update work!

**Comment by NecronomiconCoding** [comment]
we have another conflicts :laughing:

**Comment by JoshYuen** [comment]
Out to lunch atm, will get to it when I get back to a computer. I forgot my surface.

**Comment by JoshYuen** [comment]
Updated sorry was busy with work today.

**Comment by salvob41** [comment]
This is great, but calling a Session variable `ctx` is weird.

**Comment by JoshYuen** [comment]
Was done in other places as well since context was turned into Session, we can try a find and replace all.

**Comment by salvob41** [comment]
if you use a IDE, like Intellij or even Eclipse or NetBeans, there is the option of refactor variable and it will find all the variable Session with that name and refactor them. 
But find and replace might work too, be careful that aren't other variables called in that way

**Comment by salvob41** [comment]
I don't think this works. When in config `"DumpPokemonStats": true,` it doesn't generate the dumps file

**Comment by JoshYuen** [comment]
Worked before another merge. I will look into it most likely just a file path thing.

**Comment by JoshYuen** [comment]
Ah yeah I see the problem but I'm not near a computer atm. Once I get near one I wI'll resolve it. One of the file names got changed on accident. Along with the call got removed from display pokemon task from a merge.

**Comment by chopraaa** [comment]
Why was this merged before testing?

**Comment by JoshYuen** [comment]
It worked before merge and I don't think I added the display pokemon updated file to tracking when I pushed to my branch. Not an issue it doesn't break anything and when I tested it on my end it was working. It is just the display pokemon task didn't get committed.

**Comment by JoshYuen** [comment]
Ah someone already pushed a fix, thanks.

**Comment by PokemonRS** [comment]
Is it possible to add to the dump file number of candies for each type of Pokemon?

**Comment by JoshYuen** [comment]
Yes I will improve this when I have a chance.
